### DISCUSSION TITLE
Chrome Uncaught TypeError: Cannot read properties of undefined (reading '1')

### DISCUSSION THREAD

**Discussion by adamtheapiguy** [TarekRaafat/autoComplete.js#422] [General]
Hey guys,

I get the below error in chrome sometime when I click on an item in the drop-down list of autoComplete , any ideas?

<img width="1520" alt="Screenshot 2024-01-23 at 5 33 21 am" src="https://github.com/TarekRaafat/autoComplete.js/assets/11493202/bebb4e51-d6bd-4161-a6d4-8f161bbb6e6a">

Error Text:
========
autoComplete.js:333 Uncaught TypeError: Cannot read properties of undefined (reading '1')
    at feedback (autoComplete.js:333:28)
    at select (autoComplete.js:413:5)
    at click (autoComplete.js:422:7)
    at HTMLUListElement.click$1 (autoComplete.js:508:11)

**Comment by HeavenlyEntity** [comment]
Getting the same type of error. I even type cast the number to  make sure its an integer but it still getting the same error.

**Reply by AdamAliAu** [reply]
In my case I found out that the response was not as expected. Was getting 429 errors. After fixing that all went well. You need to make sure in that call, the response was returned as expected.

**Comment by sanjay-tiwari-tfs** [comment]
Even I am also facing same issue, can anyone have any idea ?

**Reply by adamtheapiguy** [reply]
My issue was that json response structure was different than what I defined as it was error response. If you check your response you might see it is not what you defined in the config and that's why you get this error.

**Reply by sanjay-tiwari-tfs** [reply]
Hi @adamtheapiguy , as per our requirement we have modified json response to show some different design, can you please suggest how to fix this issue.

**Reply by adamtheapiguy** [reply]
I think The response need to match plugin config. Response cannot be dynamic for success and fail. So for example if you configured $.result for plugin to pick up the data, all responses (success or fail) must have $.result. If fail response has no $.result then you would get that error. 

In my case I was getting the error when the response was 429 (too many requests) as this response has no data and the json path $.result was not in the response. 

To fix it, I just increased the rate limit on the api to avoid 429 errors and configured plugin to slow down the speed of api call. 

At least that was my experience. You might have a different issue.

**Comment by TarekRaafat** [comment]
Hello @adamtheapiguy, @HeavenlyEntity, and @sanjay-tiwari-tfs,

Apologies for my very delayed reply.

This error happens when `ctx.feedback.results` is undefined at the time of selection. It traces back to a lifecycle issue where the internal state is not properly populated before the click handler fires.

This is now tracked in the roadmap and will be addressed in the upcoming releases:
- #455 (data source return validation)
- #482 (re-init teardown and select fix)

Feel free to vote on those issues if they matter to you.

Thanks for reporting, and sorry again for the wait.

Have a nice day, cheers! :)
